cli.go

```diff
@@ -500,100 +490,31 @@ func (i *CLI) heartbeatCheck(heartbeatURL, heartbeatAuthToken string) error {
 
 func (i *CLI) setupHTTPAPI() {
 	i.logger.Info("setting up HTTP API")
-	http.HandleFunc("/worker", i.httpAPI)
-	http.HandleFunc("/worker/", i.httpAPI)
-}
-
-func (i *CLI) httpAPI(w http.ResponseWriter, req *http.Request) {
-	if req.Method == "GET" && req.URL.Path == "/worker" {
-		w.Header().Set("Content-Type", "text/plain;charset=utf-8")
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprintf(w, strings.TrimSpace(`
-Available methods:
-
-- POST /worker/graceful-shutdown
-- POST /worker/graceful-shutdown-pause
-- POST /worker/info
-- POST /worker/pool-decr
-- POST /worker/pool-incr
-- POST /worker/shutdown
-		`)+"\n")
-		return
-	}
-
-	if req.Method != "POST" {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	username, password, ok := req.BasicAuth()
-	if !ok {
-		w.Header().Set("WWW-Authenticate", "Basic realm=\"travis-ci/worker\"")
-		w.WriteHeader(http.StatusUnauthorized)
-		return
+	apiHandler := &APIHandler{
+		pool:       i.ProcessorPool,
+		auth:       i.c.String("http-api-auth"),
+		workerInfo: i.workerInfo,
+		cancel:     i.cancel,
 	}
+	apiHandler.Setup()
```

Comment:
I'm sorry for not catching it before, but I realize this `apiHandler` is not retained as a reference, which maybe doesn't matter (???) but it's definitely confusing for me when looking at it now. Would it be handy to keep a reference, a la `i.apiHandler`?

Reply:
It should be fine, because the handler methods should keep a reference to the struct, and the HTTP serve mux is holding a reference to those handlers.

We could store it on the `CLI` if you think that's less confusing to a new reader of the code, but I don't think we'll use the reference anywhere later.

Reply:
Sorry sorry, I should have been clear that I wasn't worried about GC, but human readers. Given there's no need for a reference, WDYT about skipping an lvar entirely like `(&Thing{}).DoStuff()`?